#!/usr/bin/env python
PACKAGE = "cras_cpp_common"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("publish_each_filter", bool_t, 0, "Whether to publish data from each filter or not.", default=False)
gen.add("max_age", double_t, 0, "Max age of messages that can be processed. Older messages are thrown away.", default=3600.0, min=0.001, max=3600.0)
gen.add("disabled_filters", str_t, 0, "Comma-separated list of filters that should be temporarily disabled.", default="")

exit(gen.generate(PACKAGE, "cras_cpp_common", "FilterChain"))
